common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -6322,29 +6322,67 @@ public ZooKeeperHiveHelper getZKConfig() {
       .trustStorePassword(trustStorePassword).build();
   }
 
+  public static HiveConf create() {
+    return new HiveConf();
+  }
+
+  public static HiveConf create(Class<?> cls) {
+    return new HiveConf(cls);
+  }
+
+  public static HiveConf create(Configuration other, Class<?> cls) {
+    return new HiveConf(other, cls);
+  }
+
+
+  public static HiveConf create(HiveConf other) {
+    return new HiveConf(other);
+  }
+
+  /**
+   * Instantiating HiveConf is deprecated. Please use
+   * HiveConf#create() to construct a Configuration,
+   * this method will become private eventually.
```

Comment:
If we make those private then we are breaking inheritance which is fine by me but wanted to mention it just in case.

For sanity reasons I would also attempt to run the precommit tests with all the constructors private to see if there are any hidden reflection calls or other exotic usages.

Reply:
yeah, I was thinking about the same, and just for curiosity's sake: if nothing is broken with private (or protected) constructors, can we make the public -> private/protected in this PR?

Reply:
There shouldn't be many projects creating new Hive configuration instances so it may be fine doing it here as part of this change. 

Consider though if we would like to backport the change in other Hive branches (Apache or internal); there breaking changes may not be tolerable.